server/src/main/java/org/opensearch/indices/replication/common/CopyState.java

```diff
@@ -58,21 +51,12 @@ public CopyState(ReplicationCheckpoint requestedReplicationCheckpoint, IndexShar
         this.infosBytes = buffer.toArrayCopy();
     }
 
-    @Override
-    protected void closeInternal() {
-        try {
-            segmentInfosRef.close();
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-    }
-
     public ReplicationCheckpoint getCheckpoint() {
         return replicationCheckpoint;
     }
 
     public Map<String, StoreFileMetadata> getMetadataMap() {
-        return metadataMap;
+        return replicationCheckpoint.getMetadataMap();
```

Comment:
Looking at this, it seems like this is no longer fetching the indexShard's metadatamap but instead the checkpoint's metadata map. Are these the same?

Reply:
Hey @scrawfor99 thanks for taking a look.  Yes these will be the same.  The metadata is recomputed post every refresh with the new set of segments and stored with its checkpoint.  The change to this class will ensure we aren't recomputing the metadata every time we construct this class & fetch the metadata from the most recently refreshed on checkpoint.